catboost/libs/eval_result/eval_helpers.cpp

```diff
@@ -217,6 +217,13 @@ void PrepareEval(const EPredictionType predictionType,
                 }
             }
             break;
+        case EPredictionType::Poisson:
+            result->resize(1);
+            (*result)[0].reserve(approx.size());
+            for (const double prediction : approx[0]) {
+                (*result)[0].push_back(exp(prediction));
+            }
```

Comment:
It's better to use [this](https://github.com/catboost/catboost/blob/master/catboost/libs/model/eval_processing.h#L15) function here. It is much faster.